store/mongo/datastore_mongo.go

```diff
@@ -265,3 +269,38 @@ func (db *DataStoreMongo) CreateIntegration(
 
 	return err
 }
+
+func (db *DataStoreMongo) GetDevice(ctx context.Context, deviceID string) (*model.Device, error) {
+	return db.GetDeviceByIntegrationID(ctx, deviceID, uuid.Nil)
```

Comment:
This function is not doing what it is intended for because deviceID is unused.

Reply:
right, sorry, I fixed it!